mitodl/mit-open
# Very minimal homepage

#### What are the relevant tickets?
#27 , but not quite all of what we discuss (no cards yet).

#### What's this PR do?
This PR:
- adds `react-query` hooks for our generated API client
- adds new data factories based on the types from our generated API client
- adds some tests using the new API client & data factories
- Adds an **extremely** simple homepage. *More coming*. 

#### How should this be manually tested?
1. If you don't have any learning resources yet, make some fake ones. In a django shell, run:
    ```py
    from learning_resources.factories import *
    CourseFactory.create_batch(4)
    ProgramFactory.create_batch(3)
    ```
2. Visit http://localhost:8063 to see a list of learning resources. 

### Underwhelming Screenshot
*This is a list of fake resources titles generated as above.*
<img width="548" alt="Screenshot 2023-08-18 at 4 36 43 PM" src="https://github.com/mitodl/mit-open/assets/9010790/a64c63dd-c08e-4b70-b8d5-a83f932e284e">
